EIPS/eip-9999.md

```diff
@@ -0,0 +1,447 @@
+---
+eip: 9999
+title: Payload Chunking with Chunk Access Lists
+description: Semantic block chunking with separated state diffs for streaming validation and reduced latency
+author: Toni Wahrstätter (@nerolation), Milos Stankovic (@morph-dev), Jihoon Song (@jihoonsong), Bharath Vedartham (@bharath-123), Raúl Kripalani (@raulk)
+discussions-to: https://ethereum-magicians.org/t/eip-9999-payload-chunking/27085
+status: Draft
+type: Standards Track
+category: Core
+created: 2025-12-01
+requires: 4844, 7732, 7928
+---
+
+## Abstract
+
+This EIP introduces Payload Chunking, a protocol upgrade that restructures Ethereum block propagation into self-contained execution chunks with separated Chunk Access Lists (CALs). Proposer propagates beacon and execution block headers, followed by CALs and Chunks. CALs propagate separately and contain state diffs that enable independent chunk execution. Chunks are executed and verified separately (as long as CALs `0..N` are available for chunk `N`). After all chunks execute, the CL validates and finalizes the block. This architecture transforms block validation from a monolithic operation into a streaming pipeline while preserving block atomicity. The concept of chunks is used only during block propagation.
+
+## Motivation
+
+As Ethereum's gas limits increase, block sizes and execution complexity grow correspondingly. The current monolithic block structure requires validators to download the entire block before beginning execution, creating a critical bottleneck in the consensus timeline. This latency becomes increasingly problematic as larger blocks take longer to propagate and decompress before execution can begin.
+
+This proposal addresses these issues by:
+
+- Enabling streaming validation where execution begins as chunks arrive
+- Supporting independent chunk execution using CALs containing state diffs
+- Bounding resource requirements to per-chunk maximums (`2**24` gas)
+- Allowing early rejection of invalid blocks without processing all chunks
+- Creating natural boundaries for future proving systems
+
+## Specification
+
+### Constants
+
+```python
+CHUNK_GAS_LIMIT = 2**24  # From EIP-7825 (16,777,216 gas per chunk)
+MAX_CHUNKS_PER_BLOCK = 2**8  # Maximum chunks in a block (256 chunks, allowing 4G gas blocks)
+MAX_TRANSACTIONS_PER_CHUNK = 2**10 # Maximum number of transactions in a chunk (1,024 tx per chunk)
```

Comment:
This is fine (for now) but if we lower intrinsic tx gas limit below `CHUNK_GAS_LIMIT // MAX_TRANSACTIONS_PER_CHUNK` then the requirement that the txs per chunk are equal to or lower than the maximum will not hold. E.g. if https://eips.ethereum.org/EIPS/eip-2780 ships.